subworkflows/local/bam_downsample/main.nf

```diff
@@ -61,25 +57,30 @@ workflow BAM_DOWNSAMPLE {
     ch_bam_emul = SAMTOOLS_VIEW.out.bam
         .combine(SAMTOOLS_INDEX_1.out.bai, by:0)
 
-    SAMTOOLS_MERGE(
-        ch_bam_emul
-            .map{
-                metaICRD, bam, index -> [metaICRD.subMap("id", "depth"), bam, index]
-            }
-            .groupTuple()
-            .map{ metaID, bam, index ->
-                [ metaID + ["chr": "all"], bam, index ]
-            },
-        ch_fasta
-    )
-    SAMTOOLS_INDEX_2(SAMTOOLS_MERGE.out.bam)
-    ch_versions = ch_versions.mix(SAMTOOLS_INDEX_2.out.versions.first())
+    if (params.sim_by_reg == true) {
+        SAMTOOLS_MERGE(
+            ch_bam_emul
+                .map{
+                    metaICRD, bam, index -> [metaICRD.subMap("id", "depth"), bam, index]
+                }
+                .groupTuple()
+                .map{ metaID, bam, index ->
+                    [ metaID + ["chr": "all"], bam, index ]
+                },
+            ch_fasta
+        )
+        ch_versions = ch_versions.mix(SAMTOOLS_MERGE.out.versions.first())
+
+        SAMTOOLS_INDEX_2(SAMTOOLS_MERGE.out.bam)
+        ch_versions = ch_versions.mix(SAMTOOLS_INDEX_2.out.versions.first())
 
-    ch_bam_emul_all = SAMTOOLS_MERGE.out.bam
-        .combine(SAMTOOLS_INDEX_2.out.bai, by:0)
+        ch_bam_emul_all = SAMTOOLS_MERGE.out.bam
+            .combine(SAMTOOLS_INDEX_2.out.bai, by:0)
+    } else {
+        ch_bam_emul_all = ch_bam_emul
```

Comment:
Similarly, is there a specific reason for adding a new channel? We could simply overwrite the `ch_bam_emul` with the `SAMTOOLS_MERGE.out.bam.combine(SAMTOOLS_INDEX_2.out.bai, by:0)`. Therefore, the else would not be needed.

I prefer that approach because it is less verbose.

Reply:
Same as before, I will ask for the best practice.